ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -929,6 +929,7 @@ let run_in_directory ~prog ~prog_is_quoted:_ ~args ~cwd ?stdin ?stdout ?stderr (
   in
   Unix.close stdin;
   if should_close_stdout then Unix.close stdout;
+  Option.iter stderr ~f:Unix.close;
```

Comment:
Are you trying to fix an issue you observed ? Looking at Merlin's codebase I don't think there is any usage of `run_in_directory` where `stderr` is not `None`.

But yes we should close that file descriptor when stderr it redirected to a file.

Reply:
Yes, I never observed this issue but I imagine it could be possible in the future.